tests/conftest.py

```diff
@@ -1460,27 +1507,41 @@ def pytest_generate_tests(metafunc):        # noqa E302
     duts_selected = None
     global _frontend_hosts_per_hwsku_per_module, _hosts_per_hwsku_per_module
     # Enumerators for duts are mutually exclusive
-    if "enum_dut_hostname" in metafunc.fixturenames:
-        duts_selected = generate_params_dut_hostname(metafunc)
-        dut_fixture_name = "enum_dut_hostname"
-    elif "enum_supervisor_dut_hostname" in metafunc.fixturenames:
-        duts_selected = generate_params_supervisor_hostname(metafunc)
-        dut_fixture_name = "enum_supervisor_dut_hostname"
-    elif "enum_frontend_dut_hostname" in metafunc.fixturenames:
-        duts_selected = generate_params_frontend_hostname(metafunc)
-        dut_fixture_name = "enum_frontend_dut_hostname"
-    elif "enum_rand_one_per_hwsku_hostname" in metafunc.fixturenames:
-        if metafunc.module not in _hosts_per_hwsku_per_module:
-            hosts_per_hwsku = generate_params_hostname_rand_per_hwsku(metafunc)
-            _hosts_per_hwsku_per_module[metafunc.module] = hosts_per_hwsku
-        duts_selected = _hosts_per_hwsku_per_module[metafunc.module]
-        dut_fixture_name = "enum_rand_one_per_hwsku_hostname"
-    elif "enum_rand_one_per_hwsku_frontend_hostname" in metafunc.fixturenames:
-        if metafunc.module not in _frontend_hosts_per_hwsku_per_module:
-            hosts_per_hwsku = generate_params_hostname_rand_per_hwsku(metafunc, frontend_only=True)
-            _frontend_hosts_per_hwsku_per_module[metafunc.module] = hosts_per_hwsku
-        duts_selected = _frontend_hosts_per_hwsku_per_module[metafunc.module]
-        dut_fixture_name = "enum_rand_one_per_hwsku_frontend_hostname"
+    target_hostname = get_target_hostname(metafunc)
```

Comment:
One of the main ideas for parallel run is to run DUT parametrizations in parallel. For example, the `snmp/test_snmp_interfaces.py` test will have the following parametrizations if we run it on a T2 chassis with multiple line cards and one supervisor card:
```
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-0]
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-lc1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-1]
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-lc3-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-2]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-2]
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-sup-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-2]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-2]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-4]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-4]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-5]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-5]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-8]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-8]
```

With this change, we can run the following 3 blocks in parallel
```
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-0]
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-lc1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-1]
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-lc3-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc1-1-2]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc1-1-2]
```

```
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-lc3-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-lc3-1-2]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-lc3-1-2]
```

```
snmp/test_snmp_interfaces.py::test_snmp_mgmt_interface[inv-sup-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-0]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-1]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-4]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-4]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-5]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-5]
snmp/test_snmp_interfaces.py::test_snmp_interfaces[inv-sup-1-8]
snmp/test_snmp_interfaces.py::test_snmp_interfaces_mibs[inv-sup-1-8]
```